subsys/bootloader/bl_validation/bl_validation.c

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <bl_validation.h>
+#include <zephyr/types.h>
+#include <errno.h>
+#include <misc/printk.h>
+#include <fw_metadata.h>
+#include <bl_crypto.h>
+#include <provision.h>
+
+bool verify_firmware(u32_t address)
+{
+	/* Some key data storage backends require word sized reads, hence
+	 * we need to ensure word alignment for 'key_data'
+	 */
+	u32_t key_data[CONFIG_SB_PUBLIC_KEY_HASH_LEN/4];
+	int retval = -EFAULT;
+	int err;
+	const struct fw_firmware_info *fw_info;
+	const struct fw_validation_info *fw_ver_info;
+
+	fw_info = fw_find_firmware_info(address);
+
+	if (!fw_info) {
+		printk("Could not find valid firmware info inside "
```

Comment:
Should we use a logger in this file instead of `printk`, as this is now a subsystem and not part of a the sample code?

Reply:
It seems to add about 10k to the b0 size, which is a bit inconvenient, so I propose we leave it for now.

Reply:
IMO size with logging is not something we should worry about unless it causes issues with building (e.g. stepping over regions and so on).

Reply:
It's non-trivial to do, so I'll change it in a later PR.